### PR TITLE
Updated to Amazon Advertising API v2.0

### PR DESCRIPTION
**Pull request by eamigo13**

### CODE REVIEW

File: lib/blurb/campaign.rb [hunk 1]
```diff
@@ -1,29 +1,30 @@
 module Blurb
   class Campaign < BaseResource
-    def self.retrieve(campaign_id)
-      get_request("/v1/campaigns/#{campaign_id}")
+    def self.retrieve(campaign_id, campaign_type)
+      get_request("/v2/#{campaign_type}/campaigns/#{campaign_id}")
     end
 
     def self.retrieve_extended(campaign_id)
```
Comment (jasonwells) [inline lib/blurb/campaign.rb]: You aren't passing `campaign_type` here, and it's used later
```suggestion
    def self.retrieve_extended(campaign_id, campaign_type)
```

File: lib/blurb/campaign.rb [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
     def self.create(params = {}, opts = {})
       # required argument checks
-      if !params["name"] && !params["campaignType"] && !params["targetingType"] && !params["state"] && !params["dailyBudget"] && !params["startDate"]
-        raise ArgumentError.new("params hash must contain name, campaignType, targetingType, state, dailyBudget and startDate")
+      if !params["name"] && !params["targetingType"] && !params["state"] && !params["dailyBudget"] && !params["startDate"]
+        raise ArgumentError.new("params hash must contain name, targetingType, state, dailyBudget and startDate")
       end
 
-      post_request("/v1/campaigns", [params])
+      post_request("/v2/sp/campaigns", [params])
```
Comment (jasonwells) [inline lib/blurb/campaign.rb]: Why does this create not take a `campaign_type` parameter, and instead always post to `/v2/sp/campaigns`?

Comment (eamigo13) [reply lib/blurb/campaign.rb]: Great Question.  The API does not support the creation of sponsored brands (formerly known as headline search ads) through the api.  Sponsored Brands must be created through the user interface so the create method can only be used for sponsored products.

File: lib/blurb/campaign.rb (hunk 1 above)
Comment (eamigo13) [reply lib/blurb/campaign.rb]: Good catch!  Thanks.

File: lib/blurb/campaign.rb (hunk 2 above)
Comment (jasonwells) [reply lib/blurb/campaign.rb]: Is this something then, where we could keep the API consistent, but throw an exception on our end if an invalid value is passed? That way it wouldn't be breaking if they ever ended up supporting it, or more, different types.

**Comment by eamigo13** [comment]
Travis CI build is failing because it won't set the private environment variables in the build spawned from a pull-request from another source.  Merging changes into master should fix the problem

**Pull request merged**